src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -560,25 +560,37 @@ internal String ToString(TraceFormat traceFormat)
                         sb.Append(']');    
                     }
 
-                    // arguments printing
-                    sb.Append('(');
-                    ParameterInfo[] pi = mb.GetParameters();
-                    bool fFirstParam = true;
-                    for (int j = 0; j < pi.Length; j++)
+#if FEATURE_CORECLR
+                    try
+                    {
+#endif
+                        // arguments printing
+                        ParameterInfo[] pi = mb.GetParameters();
+                        sb.Append('(');
+                        bool fFirstParam = true;
+                        for (int j = 0; j < pi.Length; j++)
+                        {
+                            if (fFirstParam == false)
+                                sb.Append(", ");
+                            else
+                                fFirstParam = false;
+
+                            String typeName = "<UnknownType>";
+                            if (pi[j].ParameterType != null)
+                                typeName = pi[j].ParameterType.Name;
+                            sb.Append(typeName);
+                            sb.Append(' ');
+                            sb.Append(pi[j].Name);
+                        }   
+                        sb.Append(')');
+#if FEATURE_CORECLR
+                    }
+                    catch
                     {
```

Comment:
This try/catch is pretty broad, e.g. we can end up with half of the parameter list appended. Would it be better to have the try/catch just around the call to `GetParameters()` - do not append anything if it throws, and maybe just around the call to `ParameterType` - append `<UnknownType>` as the code is trying to do if it throws?